docs/README.md

```diff
@@ -0,0 +1,640 @@
+
+⚠️ **Note:** Make sure to review and adjust the RBAC permissions as needed. The operator will create and update CRDs, so appropriate permissions should be granted. We are continuously working to determine the best way to handle this.
+
+## Configuration
+
+The Cluster API Operator uses the controller-runtime library, making it compatible with all the options that the library provides. This offers flexibility when configuring the operator and allows you to benefit from the features offered by controller-runtime.
+
+### Examples of Configuration Options
+
+Some examples of controller-runtime configuration options you can use with the Cluster API Operator include:
+
+1. **Metrics:** Controller-runtime enables you to collect and expose metrics about its internal behavior, such as the number of reconciliations executed by the operator over time. You can customize the metrics endpoint and the metrics scraping interval, among other settings.
+
+2. **Leader Election:** To ensure high availability of the operator, you can enable leader election when running multiple replicas. Controller-runtime allows you to set the leader election resource lock and polling interval to suit your needs.
+
+3. **Logger:** The operator allows you to use controller-runtime logging options to configure the logging subsystem. You can choose the logging level and output format, and even enable logging for specific libraries or components.
+
+Here's an example of how you can configure the Cluster API Operator deployment with some of these options:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: cluster-api-operator
+  namespace: capi-operator-system
+spec:
+  template:
+    spec:
+      containers:
+      - name: manager
+        args:
+        - --metrics-bind-addr=:8080
+        - --leader-elect
+        - --leader-elect-retry-period=5s
+        - --v=5
+        env:...
+```
+
+For complete details on the available configuration options, you can execute:
+
+```bash
+export CAPI_OPERATOR_VERSION=v0.3.0
+docker run -it --rm registry.k8s.io/capi-operator/cluster-api-operator:${CAPI_OPERATOR_VERSION} /manager --help
+```
+
+## Basic Cluster API Provider Installation
+
+In this section, we will walk you through the basic process of installing Cluster API providers using the operator. The Cluster API operator manages four types of objects:
+
+- CoreProvider
+- BootstrapProvider
+- ControlPlaneProvider
+- InfrastructureProvider
+
+Please note that this example provides a basic configuration of Azure Infrastructure provider for getting started. More detailed examples and CRD descriptions will be provided in subsequent sections of this document.
+
+### Installing the CoreProvider
+
+The first step is to install the CoreProvider, which is responsible for managing the Cluster API CRDs and the Cluster API controller.
+
+*Example:*
+
+```yaml
+apiVersion: operator.cluster.x-k8s.io/v1alpha1
+kind: CoreProvider
+metadata:
+  name: cluster-api
+  namespace: capi-system
```

Comment:
this ns has to be already created otherwise it would fail, can we mention/append it somewhere?

Reply:
Thank you for review! I addressed that part